packages/backend/src/open_payments/client/service.ts

```diff
@@ -0,0 +1,70 @@
+import axios from 'axios'
+import { OpenAPI, HttpMethod, ValidateFunction } from 'openapi'
+
+import { BaseService } from '../../shared/baseService'
+import { IncomingPaymentJSON } from '../payment/incoming/model'
+
+const REQUEST_TIMEOUT = 5_000 // millseconds
+
+export interface OpenPaymentsClientService {
+  incomingPayment: {
+    get(url: string): Promise<IncomingPaymentJSON | undefined>
+  }
+}
+
+export interface ServiceDependencies extends BaseService {
+  accessToken: string
+  openApi: OpenAPI
+  validateResponse: ValidateFunction<IncomingPaymentJSON>
+}
+
+export async function createOpenPaymentsClientService(
+  deps_: Omit<ServiceDependencies, 'validateResponse'>
+): Promise<OpenPaymentsClientService> {
+  const log = deps_.logger.child({
+    service: 'OpenPaymentsClientService'
+  })
+  const validateResponse =
+    deps_.openApi.createResponseValidator<IncomingPaymentJSON>({
+      path: '/incoming-payments/{incomingPaymentId}',
+      method: HttpMethod.GET
+    })
```

Comment:
Why does that have to be overwritten?

Reply:
`validateResponse` isn't included on `deps_`.
Or are you asking why define a response validator that should already exist in a validator middleware?

Reply:
Yes, the latter.

Reply:
:thinking:
Nothing is being overwritten.
Both this and the auth middleware (only for the get incoming payment route) will each create the same response validator once.
However, the auth middleware does not create response validators when running in production:
https://github.com/interledger/rafiki/blob/main/packages/openapi/src/middleware.ts#L11-L12

Reply:
Never mind, I misread.